tests/Form/Extension/ChoiceTypeExtensionTest.php

```diff
@@ -24,6 +24,10 @@ class ChoiceTypeExtensionTest extends TestCase
 {
     protected function setup(): void
     {
+        if (!class_exists(DependencyInjectionExtension::class)) {
+            $this->markTestIncomplete(sprintf('%s class was removed.', DependencyInjectionExtension::class));
+        }
```

Comment:
It would be nice to rewrite this test for cases when there is no SonataCoreBundle.

Reply:
Yes, that class was only used on sf 2.8, so we should rewrite this to not needing to declare it

Reply:
Maybe you can simply remove the usage and see if tests passes?

Reply:
The `ChoiceTypeExtension` class is not used explicitly. It only [registers as a service](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/config/form_types.xml#L40-L42). If i delete `ChoiceTypeExtension` class and its test, then the tests will be green, but will everything work for clients?

Reply:
I mean delete the usage of `DependencyInjectionExtension`

Reply:
If remove `DependencyInjectionExtension`, then the test will break. This test checks how the `ChoiceTypeExtension` is applied when working through the `FormFactoryInterface`. To remove `DependencyInjectionExtension`, we will have to use some kind of analogue.

Reply:
@jordisala1991 i managed to fix the test. Now he works for two cases when there is `DependencyInjectionExtension`, and when he is not. Is it necessary to leave the case when there is no `DependencyInjectionExtension`?